Python/Product/PythonTools/PythonTools/PythonClassifier.cs

```diff
@@ -19,58 +19,46 @@
 using System.Diagnostics;
 using System.Diagnostics.Contracts;
 using System.Text;
+using System.Threading.Tasks;
 using Microsoft.PythonTools.Editor;
 using Microsoft.PythonTools.Intellisense;
 using Microsoft.PythonTools.Interpreter;
 using Microsoft.PythonTools.Parsing;
 using Microsoft.VisualStudio.Text;
 using Microsoft.VisualStudio.Text.Classification;
+using Microsoft.VisualStudio.Text.Editor;
 
 namespace Microsoft.PythonTools {
     /// <summary>
     /// Provides classification based upon the DLR TokenCategory enum.
     /// </summary>
-    internal sealed class PythonClassifier : IClassifier {
+    internal sealed class PythonClassifier : IClassifier, IPythonTextBufferInfoEventSink {
         private readonly TokenCache _tokenCache;
         private readonly PythonClassifierProvider _provider;
-        private readonly PythonTextBufferInfo _buffer;
-        private PythonLanguageVersion _version;
 
         [ThreadStatic]
         private static Dictionary<PythonLanguageVersion, Tokenizer> _tokenizers;    // tokenizer for each version, shared between all buffers
 
-        internal PythonClassifier(PythonClassifierProvider provider, PythonTextBufferInfo buffer) {
-            buffer.OnChanged += BufferChanged;
-            buffer.OnContentTypeChanged += BufferContentTypeChanged;
-
+        internal PythonClassifier(PythonClassifierProvider provider) {
             _tokenCache = new TokenCache();
             _provider = provider;
```

Comment:
>provider [](start = 24, length = 8)

You could check for null

Reply:
It's an internal constructor that gets passed `this` - it'll never be null.